File: src/main/java/tn/esprit/nafseyti/models/rendez_vous.java
package tn.esprit.nafseyti.models;
import java.time.LocalDate;
import java.time.LocalTime;
import java.util.Objects;

public class rendez_vous {

    private int id;
    private int userId;
    private int medecinId;
    private LocalDate dateRendezVous;
    private LocalTime heureDebut;
    private LocalTime heureFin;
    private String typeSeance;
    private String statut;

    public rendez_vous() {}

    public rendez_vous(int id, int userId, int medecinId, LocalDate dateRendezVous, LocalTime heureDebut, LocalTime heureFin, String typeSeance, String statut) {
        this.id = id;
        this.userId = userId;
        this.medecinId = medecinId;
        this.dateRendezVous = dateRendezVous;
        this.heureDebut = heureDebut;
        this.heureFin = heureFin;
        this.typeSeance = typeSeance;
        this.statut = statut;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public int getMedecinId() {
        return medecinId;
    }

    public void setMedecinId(int medecinId) {
        this.medecinId = medecinId;
    }

    public LocalDate getDateRendezVous() {
        return dateRendezVous;
    }

    public void setDateRendezVous(LocalDate dateRendezVous) {
        this.dateRendezVous = dateRendezVous;
    }

    public LocalTime getHeureDebut() {
        return heureDebut;
    }

    public void setHeureDebut(LocalTime heureDebut) {
        this.heureDebut = heureDebut;
    }

    public LocalTime getHeureFin() {
        return heureFin;
    }

    public void setHeureFin(LocalTime heureFin) {
        this.heureFin = heureFin;
    }

    public String getTypeSeance() {
        return typeSeance;
    }

    public void setTypeSeance(String typeSeance) {
        this.typeSeance = typeSeance;
    }

    public String getStatut() {
        return statut;
    }

    public void setStatut(String statut) {
        this.statut = statut;
    }

    @Override
    public String toString() {
        return "rendez_vous{" +
                "id=" + id +
                ", userId=" + userId +
                ", medecinId=" + medecinId +
                ", dateRendezVous=" + dateRendezVous +
                ", heureDebut=" + heureDebut +
                ", heureFin=" + heureFin +
                ", typeSeance='" + typeSeance + '\'' +
                ", statut='" + statut + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof rendez_vous that)) return false;
        return id == that.id && userId == that.userId && medecinId == that.medecinId && Objects.equals(dateRendezVous, that.dateRendezVous) && Objects.equals(heureDebut, that.heureDebut) && Objects.equals(heureFin, that.heureFin) && Objects.equals(typeSeance, that.typeSeance) && Objects.equals(statut, that.statut);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, userId, medecinId, dateRendezVous, heureDebut, heureFin, typeSeance, statut);
    }
}

File: src/Entity/RendezVous.php
<?php

namespace App\Entity;

use App\Repository\RendezVousRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: RendezVousRepository::class)]
#[ORM\Table(name: 'rendez_vous')]
class RendezVous
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(name: 'dateRendezVous', type: 'date')]
    private ?\DateTimeInterface $dateRendezVous = null;

    #[ORM\Column(name: 'heureDebut', type: 'string', length: 10, nullable: true)]
    private ?string $heureDebut = null;

    #[ORM\Column(name: 'heureFin', type: 'string', length: 10, nullable: true)]
    private ?string $heureFin = null;

    #[ORM\Column(name: 'type_seance', type: 'string', length: 50, nullable: true)]
    private ?string $typeSeance = null;

    #[ORM\Column(name: 'statut', type: 'string', length: 20, nullable: true)]
    private ?string $statut = 'en_attente';

    #[ORM\Column(name: 'google_event_id', type: 'string', length: 255, nullable: true)]
    private ?string $googleEventId = null;

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'rendezVous')]
    #[ORM\JoinColumn(name: 'userId', referencedColumnName: 'id', nullable: true)]
    private ?User $user = null;

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'rendezVousComeMedecin')]
    #[ORM\JoinColumn(name: 'medecinId', referencedColumnName: 'id', nullable: false)]
    private ?User $medecin = null;

    #[ORM\OneToOne(mappedBy: 'rendezVous', targetEntity: FicheConsultation::class)]
    private ?FicheConsultation $ficheConsultation = null;

    public function getId(): ?int { return $this->id; }

    public function getDateRendezVous(): ?\DateTimeInterface { return $this->dateRendezVous; }
    public function setDateRendezVous(\DateTimeInterface $dateRendezVous): self { $this->dateRendezVous = $dateRendezVous; return $this; }

    public function getHeureDebut(): ?string { return $this->heureDebut; }
    public function setHeureDebut(?string $heureDebut): self { $this->heureDebut = $heureDebut; return $this; }

    public function getHeureFin(): ?string { return $this->heureFin; }
    public function setHeureFin(?string $heureFin): self { $this->heureFin = $heureFin; return $this; }

    public function getTypeSeance(): ?string { return $this->typeSeance; }
    public function setTypeSeance(?string $typeSeance): self { $this->typeSeance = $typeSeance; return $this; }

    public function getStatut(): ?string { return $this->statut; }
    public function setStatut(?string $statut): self { $this->statut = $statut; return $this; }

    public function getGoogleEventId(): ?string { return $this->googleEventId; }
    public function setGoogleEventId(?string $googleEventId): self { $this->googleEventId = $googleEventId; return $this; }

    public function getUser(): ?User { return $this->user; }
    public function setUser(?User $user): self { $this->user = $user; return $this; }

    public function getMedecin(): ?User { return $this->medecin; }
    public function setMedecin(?User $medecin): self { $this->medecin = $medecin; return $this; }

    public function getFicheConsultation(): ?FicheConsultation { return $this->ficheConsultation; }
    public function setFicheConsultation(?FicheConsultation $ficheConsultation): self { $this->ficheConsultation = $ficheConsultation; return $this; }
}